obj[] points
bool got_stuff
